src/controller/mir_to_flow.rs

```diff
@@ -739,6 +752,35 @@ pub(crate) fn make_project_node(
     FlowNode::New(n)
 }
 
+pub(crate) fn make_distinct_node(
+    name: &str,
+    parent: MirNodeRef,
+    columns: &[Column],
+    group_by: &Vec<Column>,
+    mig: &mut Migration,
+) -> FlowNode {
+    let parent_na = parent.borrow().flow_node_addr().unwrap();
+    let column_names = column_names(columns);
+
+    let group_by_indx = if group_by.is_empty() {
+        // no query parameters, so we index on the first column
+        vec![0 as usize]
```

Comment:
Shouldn't the default be to be distinct by *all* columns?

Reply:
Agree, all columns should be the default here.